### PR TITLE
cleanup: delete unused function NewNodeInfoLister

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/fake/listers.go [hunk 1]
```diff
@@ -257,10 +257,10 @@ func (nodes NodeInfoLister) HavePodsWithRequiredAntiAffinityList() ([]*framework
 // NewNodeInfoLister create a new fake NodeInfoLister from a slice of v1.Nodes.
 func NewNodeInfoLister(nodes []*v1.Node) framework.NodeInfoLister {
 	nodeInfoList := make([]*framework.NodeInfo, len(nodes))
```
Comment (denkensk) [inline pkg/scheduler/framework/fake/listers.go:259]:
```suggestion
	nodeInfoList := make([]*framework.NodeInfo, 0, len(nodes))
```

Comment (Abirdcfly) [reply pkg/scheduler/framework/fake/listers.go:259]: Let's do the same this time.Thanks.

**Review by alculquicondor — COMMENTED** [review]
It looks like the function (and probably the type too) is unused. Please remove it instead.

**Comment by Abirdcfly** [comment]
> It looks like the function (and probably the type too) is unused. Please remove it instead.

I have thought about deleting this function directly, but it is a public function and there may be other projects that are calling this function, will there be any compatibility issues if I delete this function directly?

**Comment by alculquicondor** [comment]
I didn't find any usages in cluster-autoscaler, descheduler or scheduler-plugins. kubernetes/kubernetes is not supposed to be a library, so we don't care about keeping exported functions that we (kubernetes projects is general) are not using.

**Comment by Abirdcfly** [comment]
> I didn't find any usages in cluster-autoscaler, descheduler or scheduler-plugins. kubernetes/kubernetes is not supposed to be a library, so we don't care about keeping exported functions that we (kubernetes projects is general) are not using.

I agree and the code has been updated.😊

**Comment by Abirdcfly** [comment]
> Can you remove the NodeInfoLister type as well?

I'm afraid not, there is [some code](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/extender_test.go#L528) that uses this type.

**Comment by alculquicondor** [comment]
Sg, thanks

/lgtm

**Pull request merged**